test/e2e/apps/rc.go

```diff
@@ -84,6 +97,163 @@ var _ = SIGDescribe("ReplicationController", func() {
+			}
+		}
+		framework.ExpectEqual(rcScale.Status.Replicas, testRcMaxReplicaCount, "ReplicationController ReplicasSet Scale does not match the expected scale")
+
+		// Get the ReplicationController
+		rc, err := f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Get(context.TODO(), testRcName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Failed to fetch ReplicationController")
+		framework.ExpectEqual(rc.ObjectMeta.Labels["test-rc"], "patched", "ReplicationController is missing a label from earlier patch")
+
+		rcStatusUpdatePayload := rc
+		rcStatusUpdatePayload.Status.AvailableReplicas = 1
+		rcStatusUpdatePayload.Status.ReadyReplicas = 1
+
+		// Replace the ReplicationController's status
+		rcStatus, err = f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).UpdateStatus(context.TODO(), rcStatusUpdatePayload, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "Failed to update ReplicationControllerStatus")
+		framework.ExpectEqual(rcStatus.Status.ReadyReplicas, int32(1), "ReplicationControllerStatus readyReplicas does not equal 1")
+
+		for event := range rcWatchChan {
+			rc, ok := event.Object.(*v1.ReplicationController)
+			framework.ExpectEqual(ok, true, "Unable to convert type of ReplicationController watch event")
+			if rc.Status.Replicas == testRcMaxReplicaCount && rc.Status.ReadyReplicas == testRcMaxReplicaCount {
+				break
+			}
+		}
+
+		rcs, err := f.ClientSet.CoreV1().ReplicationControllers("").List(context.TODO(), metav1.ListOptions{LabelSelector: "test-rc-static=true"})
+		framework.ExpectNoError(err, "Failed to list ReplicationController")
+		framework.ExpectEqual(len(rcs.Items) > 0, true)
+
+		foundRc := false
+		for _, rcItem := range rcs.Items {
+			if rcItem.ObjectMeta.Name == testRcName &&
+				rcItem.ObjectMeta.Namespace == testRcNamespace &&
+				rcItem.ObjectMeta.Labels["test-rc-static"] == "true" &&
+				rcItem.ObjectMeta.Labels["test-rc"] == "patched" &&
+				rcItem.Status.Replicas == testRcMaxReplicaCount &&
+				rcItem.Status.ReadyReplicas == testRcMaxReplicaCount {
+				foundRc = true
+			}
+		}
+		framework.ExpectEqual(foundRc, true)
+
+		// Delete ReplicationController
+		err = f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).DeleteCollection(context.TODO(), &metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "test-rc-static=true"})
+		framework.ExpectNoError(err, "Failed to delete ReplicationControllers")
+
+		for event := range rcWatchChan {
+			rc, ok := event.Object.(*v1.ReplicationController)
+			framework.ExpectEqual(ok, true, "Unable to convert type of ReplicationController watch event")
+			if rc.ObjectMeta.DeletionTimestamp != nil {
+				break
+			}
+		}
+		time.Sleep(1 * time.Second)
+
+		// Get the ReplicationController to check that it's deleted
+		_, err = f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).Get(context.TODO(), testRcName, metav1.GetOptions{})
+		framework.ExpectError(err, "Failed to delete ReplicationController")
```

Comment:
I guess we should expect 404 error to verify the deletion of the ReplicationController.

Reply:
That's the idea, per https://github.com/kubernetes/kubernetes/pull/88034#discussion_r381463433, which is the functionality that this checks

Reply:
Is this checking the error text? We can't really do that in conformance tests since those are not subject to any deprecation policy. Can we check the error code?

Reply:
No it isn't. The second field in `framework.ExpectError` is a message that's displayed along with the error, which is only printed if `err` is not equal to `nil`